### PR TITLE
Vehicle update2 Part 1d: Vehicle Driving Sounds Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Continuing from #738.

### CODE REVIEW

**Review by AaronVanGeffen — APPROVED** [review]
Code looks logical. I also don't hear any abnormalities while running this.

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 1]
```diff
@@ -371,9 +372,276 @@ namespace OpenLoco::Vehicles
...
+            }
+            else
+            {
+                targetFrequency = snd->var_04;
+                targetVolume = snd->var_06;
+            }
+        }
+        else if (vehType2_2->var_5A == 1)
+        {
+            if (vehType2or6->isVehicle2() || train.cars.firstCar.front->var_5E == 0)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:476]: `!isVehicle2()`

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 2]
```diff
@@ -371,9 +372,276 @@ namespace OpenLoco::Vehicles
...
+        }
+        else
+        {
+            targetFrequency = snd->defaultFrequency;
+            targetVolume = snd->defaultVolume;
+        }
+
+        if (var5aEqual1Code == true)
+        {
+            if (vehType2or6->isVehicle2() || train.cars.firstCar.front->var_5E == 0)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:581]: `!isVehicle2()`

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 3]
```diff
@@ -371,9 +372,276 @@ namespace OpenLoco::Vehicles
...
+                    {
+                        frequency -= snd->thirdGearFreqFactor;
+                        if (speed >= snd->thirdGearSpeed)
+                        {
+                            frequency -= snd->fourthGearFreqFactor;
+                        }
+                    }
+                }
+                speed <<= 16;
+                targetFrequency = (uint16_t)((speed >> snd->speedFreqFactor) + frequency);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:600]: static_cast

File: src/OpenLoco/Objects/VehicleObject.h [hunk 4]
```diff
@@ -39,52 +39,52 @@ namespace OpenLoco
...
-        uint8_t var_10;
+        uint16_t freqIncreaseStep;  // 0xA
+        uint16_t freqDecreaseStep;  // 0xC
+        uint8_t volumeIncreaseStep; // 0xE
+        uint8_t volumeDecreaseStep; // 0xF
+        uint8_t speedFreqFactor;    // 0x10
     };
 
-    struct vehicle_object_sound_3
+    struct VehicleObjectSoundEngine2
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Objects/VehicleObject.h:69]: I get that you'd probably like to keep the `VehicleObjectSound` part as a base, but now it reads like `Sound Engine` instead. I suggest:
```suggestion
    struct VehicleObjectEngineSound2
```

**Pull request merged**